api/address/routes.go

```diff
@@ -357,8 +372,104 @@ func GetESDTBalance(c *gin.Context) {
 
 	tokenData := esdtTokenData{
 		TokenIdentifier: tokenIdentifier,
-		Balance:         balance,
-		Properties:      freeze,
+		Balance:         esdtData.Value.String(),
+		Properties:      string(esdtData.Properties),
+	}
+
+	c.JSON(
+		http.StatusOK,
+		shared.GenericAPIResponse{
+			Data:  gin.H{"tokenData": tokenData},
+			Error: "",
+			Code:  shared.ReturnCodeSuccess,
+		},
+	)
+}
+
+// GetESDTNFTData returns the nft data for the given token
+func GetESDTNFTData(c *gin.Context) {
+	facade, ok := getFacade(c)
+	if !ok {
+		return
+	}
+
+	addr := c.Param("address")
+	if addr == "" {
+		c.JSON(
+			http.StatusBadRequest,
+			shared.GenericAPIResponse{
+				Data:  nil,
+				Error: fmt.Sprintf("%s: %s", errors.ErrGetESDTNFTData.Error(), errors.ErrEmptyAddress.Error()),
+				Code:  shared.ReturnCodeRequestError,
+			},
+		)
+		return
+	}
+
+	tokenIdentifier := c.Param("tokenIdentifier")
+	if tokenIdentifier == "" {
+		c.JSON(
+			http.StatusBadRequest,
+			shared.GenericAPIResponse{
+				Data:  nil,
+				Error: fmt.Sprintf("%s: %s", errors.ErrGetESDTNFTData.Error(), errors.ErrEmptyKey.Error()),
+				Code:  shared.ReturnCodeRequestError,
+			},
+		)
+		return
+	}
+
+	nonceAsStr := c.Param("nonce")
```

Comment:
Can you make a method here `getNonceParam` where you read the nonce parameter and do the cast?

Reply:
would leave as it is.